bin/exec.ml

```diff
@@ -44,92 +191,50 @@ let term =
-          let open Memo.O in
-          let+ hints =
-            (* Good candidates for the "./x.exe" instead of "x.exe" error are
-               executables present in the current directory. Note: we do not
-               check directory targets here; even if they do indeed include a
-               matching executable, they would be located in a subdirectory of
-               [dir], so it's unclear if that's what the user wanted. *)
-            let+ candidates =
-              Build_system.files_of ~dir:(Path.build dir)
-              >>| Path.Set.to_list
-              >>| List.filter ~f:(fun p -> Path.extension p = ".exe")
-              >>| List.map ~f:(fun p -> "./" ^ Path.basename p)
-            in
-            User_message.did_you_mean prog ~candidates
-          in
-          User_error.raise ~hints [ Pp.textf "Program %S not found!" prog ]
-        in
-        let* prog =
-          let open Memo.O in
-          Build_system.run_exn (fun () ->
-              match Filename.analyze_program_name prog with
-              | In_path -> (
-                Super_context.resolve_program sctx ~dir ~loc:None prog
-                >>= function
-                | Error (_ : Action.Prog.Not_found.t) -> not_found ()
-                | Ok prog -> build_prog prog)
-              | Relative_to_current_dir -> (
-                let path =
-                  Path.relative_to_source_in_build_or_external ~dir prog
-                in
-                (Build_system.file_exists path >>= function
-                 | true -> Memo.return (Some path)
-                 | false -> (
-                   if not (Filename.check_suffix prog ".exe") then
-                     Memo.return None
-                   else
-                     let path = Path.extend_basename path ~suffix:".exe" in
-                     Build_system.file_exists path >>= function
-                     | true -> Memo.return (Some path)
-                     | false -> Memo.return None))
-                >>= function
-                | Some path -> build_prog path
-                | None -> not_found ())
-              | Absolute -> (
-                match
-                  let prog = Path.of_string prog in
-                  if Path.exists prog then Some prog
-                  else if not Sys.win32 then None
-                  else
-                    let prog = Path.extend_basename prog ~suffix:Bin.exe in
-                    Option.some_if (Path.exists prog) prog
-                with
-                | Some prog -> Memo.return prog
-                | None -> not_found ()))
-        in
-        let prog = Path.to_string prog in
-        let argv = prog :: args in
-        let env = Super_context.context_env sctx in
-        Fiber.return (prog, argv, env))
-  in
-  restore_cwd_and_execve common prog argv env
+  (* This will prevent status line printing from interfering with the output of
+     the running program. *)
+  Console.Backend.set Console.Backend.dumb;
```

Comment:
I don't think this is right because we aren't calling `finish` on the backend. I'll remove it for now and this fix can come independently since it also affects normal `dune exec`.